mlx/backend/gpu/copy.h

```diff
@@ -43,4 +43,7 @@ void copy_gpu_inplace(
 // Fill the output with the scalar val
 void fill_gpu(const array& val, array& out, const Stream& s);
 
+// Return a contiguous array with same shape that copies the data of |arr|.
+array make_contiguous_array_gpu(const array& arr, const Stream& s);
```

Comment:
A bit of a nit.. wdyt about the name `contiguous_copy_gpu` instead? It kind of fits better as a copy utility. Or `make_contiguous_gpu` (having `array` in the name seems superfluous..)

Reply:
`contiguous_copy_gpu` sounds a better name, I have updated the PR.